analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -134,21 +135,38 @@ internal fun SpdxExternalDocumentReference.getSpdxPackage(packageId: String, wor
 private fun resolveExternalDocumentReference(
     externalDocumentReference: SpdxExternalDocumentReference,
     workingDir: File
-): File {
+): SpdxDocument {
     val uri = runCatching { URI(externalDocumentReference.spdxDocument) }.getOrElse {
-        throw IllegalArgumentException("'${externalDocumentReference.spdxDocument}' identified by " +
-                "${externalDocumentReference.externalDocumentId} is not a valid URI. }")
+        throw IllegalArgumentException(
+            "'${externalDocumentReference.spdxDocument}' identified by " +
+                    "${externalDocumentReference.externalDocumentId} is not a valid URI. }"
+        )
     }
 
     if (uri.scheme.equals("file", ignoreCase = true) || !uri.isAbsolute) {
         val referencedFile = workingDir.resolve(uri.path)
-        return referencedFile.takeIf { it.isFile }
-            ?: throw IllegalArgumentException("The local file URI '$uri' does not point to an existing file. ")
+        return SpdxModelMapper.read(referencedFile.takeIf { it.isFile }
+            ?: throw IllegalArgumentException("The local file URI '$uri' does not point to an existing file.")
+        )
     }
 
-    throw IllegalArgumentException("Only URIs to local files are supported for now, but '$uri' is none.")
+    return requestSpdxDocument(uri)
 }
 
+/**
+ * Returns [File] from a given [URI] if it can be found.
+ * @throws DownloadException when file can not be downloaded.
+ */
+private fun requestSpdxDocument(uri: URI): SpdxDocument =
+    OkHttpClientHelper.downloadText(uri.toString()).mapCatching {
+        val fileExtension = File(uri.path).extension
+        return if (fileExtension == "yaml" || fileExtension == "yml") {
+            SpdxModelMapper.fromYaml(it)
+        } else {
+            SpdxModelMapper.fromJson(it)
+        }
```

Comment:
Can be replaced with one-liner

    SpdxModelMapper.FileFormat.forFile(File(uri.path)).mapper.readValue<SpdxDocument>(it)

and as a result probably be inlined.

Reply:
I integrated your one-liner, but still left it in a separate function, because I like the responsibility separation more.